ibaby_smarthome_multinode/README.md

```diff
@@ -0,0 +1,306 @@
+|  PMOD WiFi          |        Provide WiFi Connection                                |
+
+### Run This Application
+
+Modify the settings for connecting to the LwM2M Server(Gateway), as shown below:
+
+(path: `src/wearable_node/function/lwm2m/lwm2m.c`):
+
+		const static char *p_port   = (char *)"5683";    /* lwm2mServer's port and IP */
+		const static char *p_server = (char *)"192.168.43.199";
+		const static char *p_client_name = (char *)"wn"; /* name of lwm2m client node */
+
+(path: `src/lamp_node/function/lwm2m/lwm2m.c`):
+
+		const static char *p_port   = (char *)"5683";    /* lwm2mServer's port and IP */
+		const static char *p_server = (char *)"192.168.43.199";
+		const static char *p_client_name = (char *)"ln"; /* name of lwm2m client node */
+
+Here take **EMSK2.2 - ARC EM11D** with Metaware Toolset for example to show how to run this application.
+
+1. We need to use embARC 2nd bootloader to automatically load application binary for different EMSK and run. See *embARC Secondary Bootloader Example* for reference.
+
+2. Open your serial terminal such as Tera-Term on PC, and configure it to right COM port and *115200bps*.
+
+3. Interact using EMSK and Freeboard.
+
+### Add a New Node
+
+Seeing **Lamp Node** for reference. It's complete and very helpful to learn how to add a new node to iBaby System although it seems very simple.
+
+|  folder/file        |            Function                                           |
+| ------------------- | ------------------------------------------------------------- |
+|  driver             |        source code of drivers                                 |
+|  function           |        source code of function modules                        |
+|  lwm2m_client       |        source code of LwM2M Client                            |
+|  FreeRTOSConfig.h   |        header file of FreeRTOS configurations                 |
+|  main.c             |        main entry of embARC Application                       |
+|  makefile           |        Makefile of embARC Application                         |
+
+#### Makefile
+
+- Selected FreeRTOS here, then you can use [FreeRTOS API][39] in your application:
+
+		# Selected OS
+		OS_SEL ?= freertos
+
+- Target options about EMSK and toolchain:
+
+		BOARD ?= emsk
+		BD_VER ?= 22
+		CUR_CORE ?= arcem11d
+		TOOLCHAIN ?= gnu
+
+- Reset the heap and stack size for LwM2M, make sure they are big enough for your application:
+
+		##
+		# HEAP & STACK SETTINGS
+		# For LwM2M Stack Usage
+		##
+		HEAPSZ ?= 81920
+		STACKSZ ?= 81920
+
+- The relative series of the root directory, here the path of the Makefile is `./embarc_osp/application/ibaby_smarthome_multinode/src/lamp_node/makefile`:
+
+		#
+		# root dir of embARC
+		#
+		EMBARC_ROOT = ../../../..
+
+- The middleware used in your application:
+
+		MID_SEL = common lwip-contrib wakaama fatfs lwip
+
+	common for baremetal function, lwip-contrib and wakaama for LwM2M, fatfs for file system, lwip for wifi.
```

Comment:
common for baremetal function, lwip-contrib and wakaama for LwM2M, fatfs for file system, lwip for wifi.

--->

`common` for baremetal function, `lwip`, `lwip-contrib` and `wakaama` for LwM2M, `fatfs` for file system .